lib/plug/parsers/multipart.ex

```diff
@@ -14,6 +14,8 @@ defmodule Plug.Parsers.MULTIPART do
       {:more, _params, state} ->
         {:error, :too_large, %{conn | adapter: {adapter, state}}}
     end
+  rescue
```

Comment:
Can we do this instead:

```elixir
    try do
      adapter.parse_req_multipart(state, opts, &handle_headers/1)
    rescue
      e -> raise Plug.Parsers.ParseError, exception: e
    else
      {:ok, params, state} ->
        {:ok, params, %{conn | adapter: {adapter, state}}}
      {:more, _params, state} ->
        {:error, :too_large, %{conn | adapter: {adapter, state}}}
    end
```

The rationale is that we don't want to wrap everything, only the multipart passing. :)